test/index.js

```diff
@@ -464,6 +464,31 @@ describe('def', function() {
                    'as its first argument; received 0.5'));
   });
 
+  it('supports enumerated types', function() {
+    //  TimeUnit :: Type
+    var TimeUnit = $.EnumType(['milliseconds', 'seconds']);
+
+    var def = $.create($.env.concat([TimeUnit, $.ValidDate, $.ValidNumber]));
+
+    //  convertTo :: TimeUnit -> ValidDate -> ValidNumber
+    var convertTo =
+    def('convertTo',
+        {},
+        [TimeUnit, $.ValidDate, $.ValidNumber],
+        function(unit, date) {
+          var multiplier = unit === 'milliseconds' ? 1000 : 1;
+          return date.valueOf() * multiplier;
```

Comment:
Okay. I'm primarily concerned with internal consistency within the source code, so this is one of the *rare* occasions when I'm not fussed. :)